src/Tooltip/tooltip.css

```diff
@@ -135,201 +199,296 @@
-
-    .tooltip
+    &Left
     {
-        right:   0;
+      .tooltipContainer
+      {
+        transform: translate3d( -95%, 10px, 0 );
+      }
     }
-}
-
-.position__topLeft,
-.position__topRight
-{
-    .tooltipContainer
+    &Right
     {
-        transform:  translateY( -100% );
+      .tooltipContainer
+      {
+        transform: translate3d( 95%, 10px, 0 );
+      }
     }
 }
 
-.position__topLeft
+.position__topLeft,
+.position__bottomLeft
 {
-    .tooltip
-    {
-        transform:  translate3d( -50%, calc( -1 * var( --tooltipOffset ) ), 0 );
-    }
+  .tooltipContainer
+  {
+    left: 50%;
+    text-align: right;
+  }
+  .tooltip
+  {
+    &::after
+    {
+      transform: translate3d( 0, calc( 1.75 * var( --tooltipOffset ) ), 0);
+      right: 14px;
+    }
+  }
 }
 
-.position__topRight
+.position__topRight,
+.position__bottomRight
 {
-    .tooltip
-    {
-        transform:  translate3d( 50%, calc( -1 * var( --tooltipOffset ) ), 0 );
-    }
+  .tooltipContainer
+  {
+    right: 50%;
+  }
+  .tooltip
+  {
+    &::after
+    {
+      transform: translate3d( 0, calc( 1.75 * var( --tooltipOffset ) ), 0);
+      left: 14px;
+    }
+  }
 }
 
 .position__bottomLeft,
 .position__bottomRight
-{
-    .tooltipContainer
-    {
-        transform:  translateY( 100% );
-    }
-}
-
-.position__bottomLeft
 {
     .tooltip
     {
-        transform:  translate3d( -50%, var( --tooltipOffset ), 0 );
+      &::after
+      {
+        transform: translate3d( 0, calc( -1.75 * var( --tooltipOffset ) ), 0);
+      }
     }
 }
 
-.position__bottomRight
-{
-    .tooltip
-    {
-        transform:  translate3d( 50%, var( --tooltipOffset ), 0 );
-    }
-}
 
 .position__left,
 .position__right
 {
     .tooltipContainer
     {
         top:    50%;
-        width:  var( --tooltipMaxWidth );
-        height: calc( 2 * var( --tooltipMaxHeight ) );
+        transform: translate3d( 0, -50%, 0);
     }
 
     .tooltip
     {
-        top:    50%;
-
         &::before
         {
             top:    0;
             bottom: 0;
             width:  var( --tooltipOffset );
         }
+        &::after
+        {
+          top:    50%;
+          transform: translate3d( calc( 1.75 * var( --tooltipOffset ) ), -50%, 0);
+        }
     }
 }
+.position__left,
+.position__leftTop,
+.position__leftBottom
+{
+  .tooltipContainer
+  {
+    text-align: right;
+  }
+}
 
 .position__left
 {
     .tooltipContainer
     {
-        left:       0;
-        transform:  translate3d( -100%, -50%, 0 );
+        right:       100%;
+        transform: translate3d( -14px, -50%, 0 );
     }
 
     .tooltip
     {
-        right:      0;
-        transform:  translate3d( calc( -1 * var( --tooltipOffset ) ), -50%, 0 );
-
         &::before
         {
             left:   100%;
         }
+        &::after
+        {
+          border-right-color: transparent;
+          right: 0;
+          transform: translate3d( calc( 1.75 * var( --tooltipOffset ) ), -50%, 0);
+        }
+    }
+    &Top
```

Comment:
If we want to use this type of nesting style to build classname selectors here then – for consistency – we should also use it elsewhere this file (e.g. for `topLeft`, `topRight` `bottomLeft`, `bottomRight`).

(I gotta say I’m generally not really a fan of this style tho. One of the creators of Sass has even said that he [sincerely regrets adding it](https://github.com/w3c/csswg-drafts/issues/2701#issuecomment-395572904). That said, I can kinda see the value of it in this case so I fine with using it for this component.)